server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -417,21 +417,13 @@ private StorageGroupProcessor getStorageGroupProcessorByPath(PartialPath deviceP
         .get(storageGroupMNode.getPartialPath());
     if (virtualStorageGroupManager == null) {
       // if finish recover
-      if (isAllSgReady.get()) {
-        waitAllSgReady(devicePath);
-        synchronized (storageGroupMNode) {
-          virtualStorageGroupManager = processorMap.get(storageGroupMNode.getPartialPath());
-          if (virtualStorageGroupManager == null) {
-            virtualStorageGroupManager = new VirtualStorageGroupManager();
-            processorMap.put(storageGroupMNode.getPartialPath(), virtualStorageGroupManager);
-          }
+      waitAllSgReady(devicePath);
```

Comment:
if sg is not ready, tell the user that message ASAP is a good design... 
In general, blocking user's call is not good